src/math_tools/mod.rs

```diff
@@ -0,0 +1,125 @@
+//! # Math Tools module
+//!
+//! This module contain some useful mathematic tools
+//! Those tools will be used for our computations
+
+use factorial::Factorial;
+
+/// This is the math sigma function
+/// It execute the given function on the given range and return the sum of the result of the function.
+///
+/// ## Parameters
+/// i (isize): The first value of the range
+///
+/// k (isize): The last value of the range
+///
+/// func (fn(isize, Option<P>, isize, isize)) -> T: The function to execute on the range.
+/// This function params are : 1. The current value of the range 2. The parameter of the function 3. The initial value of the range 4. The final value of the range
+///
+/// param (Option<P>): A parameter that can be passed to the function
+///
+/// ## Return
+/// T -> The result of our sigma computation
+pub fn sigma<T: std::ops::AddAssign, P: Copy>(
+    i: isize,
+    k: isize,
+    func: fn(isize, Option<P>, isize, isize) -> T,
+    param: Option<P>,
+) -> T {
+    let mut sum = func(i, param, i, k);
+    for j in (i + 1)..k {
+        sum += func(j, param, i, k);
+    }
+    return sum;
+}
+
+/// This function compute the bernoulli coefficient
+///
+/// ## Parameters
+/// n(usize): The number of trials
+///
+/// k(usize): The number of sucess
+///
+/// ## Return
+/// usize -> The bernouilli coefficient for our case
+fn compute_coefficient(n: usize, k: usize) -> usize {
+    let mut num: u128 = n as u128;
+    if k == 0 || n == 0 {
+        return 1;
+    }
+    for i in 1..k {
+        num *= (n - i) as u128;
+    }
```

Comment:
Seems like a Factorial for me maybe rust have a mathematics std call or crate call for that (if it's faster it's interesting else not really)